javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java

```diff
@@ -25,6 +27,12 @@ public static AutoConfiguredOpenTelemetrySdk installOpenTelemetrySdk(
     AutoConfiguredOpenTelemetrySdk autoConfiguredSdk =
         AutoConfiguredOpenTelemetrySdk.builder()
             .setResultAsGlobal(true)
+            .addMeterProviderCustomizer(
+                (builder, properties) -> {
+                  HttpMetricsViews.registerViews(builder);
+                  RpcMetricsView.registerViews(builder);
+                  return builder;
+                })
```

Comment:
I think this might be a bad idea. Somewhat surprisingly, doing this prevents a user from using view to drop the instrument or adjust its attributes. This is because when multiple views are registered match a particular instrument, there is no merging of the views. Instead, multiple metrics are produced.

To see what I mean, try configuring the SDK in the test code of this branch to add a view which drops `http.server.duration` or changes the attributes of `http.server.duration`. You should multiple metrics named `http.server.duration` in the logs, each which the configuration of one of the matching views.

Reply:
Yes I suspected the same, hence I am still thinking of how to approach this from the user perspective as I marked in the checklist.

But anyhow I believe that making views instead of hard coded filtering would still be the first step in this direction.

The current solution I have is that the user overrides the SdkMetricProvider completely instead of adding in more views over it.

@trask @mateuszrzeszutek do you have some suggestions?

Another approach I believe would be to allow user to customise and configure these views like a decorator pattern by exposing some sort of APIs.

Third approach which I am exploring is to move these configs to view config files and allows user to override these config files either by merging or replacing the config files. This is a more plausible approach but currently due to some dependency issues I am ending up with NoClassDef error.

May be the first draft won't be a cleaner solution for the user to integrate because I suspect that some changes in the base libraries would be needed as well. But it will still be better than having no solution at all like today.

@trask

Reply:
This would also introduce a breaking change for the library instrumentation users (including spring sleuth/observability): they'd now have to add views by themselves, and ensure that they views match the OTel metrics spec.

I think this should be resolved on the spec level. I've created a separate issue just for this: https://github.com/open-telemetry/opentelemetry-specification/issues/3061

I think this is one of the two known use cases for the future "hint API", and that it's nearly impossible to resolve this (and not break anything) without the necessary API changes.